### PR TITLE
Removed restangular from DeliveryServiceSslKeysService

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?
Removes a dependency on Restangular from the "DeliveryServiceSslKeysService"
- [x] This PR partially addresses #3571 

## Which Traffic Control components are affected by this PR?

- Traffic Portal

Traffic Portal dependencies are not individually documented, and so no documentation changes are necessary.

## What is the best way to verify this PR?

No functionality should have changed (except that errors will now be logged instead of ignored in many cases), so the existing tests should all pass.

Jeremy: actually, the only way to truly verify that the functionality did not change is to:

1. run the UI tests (which is a very small subset of all TP functionality)
2. manually verify the functionality associated with ALL the services functions that changed just to be sure all is well.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4012/
Test PASSed.

File: traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js [hunk 1]
```diff
@@ -29,63 +29,54 @@ var DeliveryServiceSslKeysService = function($http, $q, locationUtils, messageMo
...
 		generateSslKeyForm.key = deliveryService.xmlId;
 
-		var request = $q.defer();
-        $http.post(ENV.api['root'] + "deliveryservices/sslkeys/generate", generateSslKeyForm)
-        .then(
+        return $http.post(ENV.api['root'] + "deliveryservices/sslkeys/generate", generateSslKeyForm).then(
             function(result) {
-            	messageModel.setMessages([ { level: 'success', text: 'SSL Keys generated and updated for ' + deliveryService.xmlId } ], true);
-                request.resolve(result.data.response);
+            	messageModel.setMessages(result.data.response, true);
```
Comment (mitchell852) [inline traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js]: The response to this api call is strange therefore you can either:

a. fix the response or 
b. or change this to:

`messageModel.setMessages([ { level: 'success', text: result.data.response } ], true);`

File: traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ var DeliveryServiceSslKeysService = function($http, locationUtils, messageModel,
 
         return $http.post(ENV.api['root'] + "deliveryservices/sslkeys/generate", generateSslKeyForm).then(
             function(result) {
-            	messageModel.setMessages(result.data.response, true);
+            	messageModel.setMessages({level: "success", text: result.data.response}, true);
```
Comment (mitchell852) [inline traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js]: close. change to:

`[{level: "success", text: result.data.response}]

Comment (ocket8888) [reply traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js]: weird

**Pull request merged**